src/applications/apps.js

```diff
@@ -36,10 +35,6 @@ export function registerApplication(appName, arg1, arg2, customProps = {}) {
 
   let loadImpl, activeWhen;
   if (!arg2) {
```

Comment:
Now that setLoader will be gone, the registerApplication function will require exactly 3 arguments all the time. So when arg2 is not defined, we should throw an error instead of proceeding.

We could even write a test verifying that registerApplication requires three valid arguments.

Reply:
Cool. I'll do that.